lib/spack/spack/fetch_strategy.py

```diff
@@ -392,7 +392,26 @@ def __init__(self, *args, **kwargs):
 
     @_needs_stage
     def fetch(self):
-        super(CacheURLFetchStrategy, self).fetch()
+        path = re.sub('^file://', '', self.url)
```

Comment:
I see you also removed the "file://" prepend in FsCache.fetcher so this may not be required (though certainly doesn't hurt)